File: agent/tools/MCP/mcp-1panel/src/mcp_1panel/panel_client.py
# 1Panel API Client — 使用 API Key 认证
# 配置方式：环境变量 PANEL_HOST + PANEL_API_KEY
import os, time, hashlib, requests


class PanelClient:
    def __init__(self, host=None, api_key=None):
        self.base = (
            host or os.getenv("PANEL_HOST")
        )
        if not self.base:
            raise ValueError("PANEL_HOST 未设置（环境变量或构造参数）")
        self.base = self.base.rstrip("/") + "/api/v2"

        self.api_key = api_key or os.getenv("PANEL_API_KEY")
        if not self.api_key:
            raise ValueError("PANEL_API_KEY 未设置（环境变量或构造参数）")

    def _auth_headers(self):
        ts = str(int(time.time()))
        raw = "1panel" + self.api_key + ts
        token = hashlib.md5(raw.encode()).hexdigest()
        return {
            "Content-Type": "application/json",
            "1Panel-Token": token,
            "1Panel-Timestamp": ts,
        }

    def _request(self, method, path, **kwargs):
        url = self.base + path
        headers = self._auth_headers()
        resp = requests.request(method, url, headers=headers, timeout=15, **kwargs)
        data = resp.json()
        if data.get("code") != 200:
            msg = data.get("message", resp.text)
            raise RuntimeError(f"API 错误 [{resp.status_code}]: {msg}")
        return data.get("data")

    def get(self, path, params=None):
        return self._request("GET", path, params=params)

    def post(self, path, body=None):
        return self._request("POST", path, json=body or {})

    def delete(self, path, body=None):
        return self._request("DELETE", path, json=body or {})

    def put(self, path, body=None):
        return self._request("PUT", path, json=body or {})

File: agent/tools/MCP/npm/mcp-1panel/src/panel-client.js
/**
 * 1Panel API Client — uses API Key authentication (MD5 token signing)
 * Configuration: env vars PANEL_HOST + PANEL_API_KEY
 */
import crypto from "node:crypto";

export class PanelClient {
  constructor(host, apiKey) {
    this.base = (host || process.env.PANEL_HOST || "").replace(/\/+$/, "");
    if (!this.base) {
      throw new Error("PANEL_HOST not set (env var or constructor param)");
    }
    this.base += "/api/v2";

    this.apiKey = apiKey || process.env.PANEL_API_KEY || process.env.PANEL_ACCESS_TOKEN || "";
    if (!this.apiKey) {
      throw new Error("PANEL_API_KEY or PANEL_ACCESS_TOKEN not set (env var or constructor param)");
    }
  }

  _authHeaders() {
    const ts = String(Math.floor(Date.now() / 1000));
    const raw = "1panel" + this.apiKey + ts;
    const token = crypto.createHash("md5").update(raw).digest("hex");
    return {
      "Content-Type": "application/json",
      "1Panel-Token": token,
      "1Panel-Timestamp": ts,
    };
  }

  async _request(method, path, body) {
    const url = this.base + path;
    const headers = this._authHeaders();
    const opts = {
      method,
      headers,
    };
    if (body !== undefined) {
      opts.body = JSON.stringify(body);
    }
    const resp = await fetch(url, opts);
    if (!resp.ok) {
      throw new Error(`HTTP ${resp.status}: ${resp.statusText}`);
    }
    const data = await resp.json();
    if (data.code !== 200) {
      const msg = data.message || resp.statusText;
      throw new Error(`API error [${resp.status}]: ${msg}`);
    }
    return data.data;
  }

  async get(path, params) {
    let url = path;
    if (params) {
      const qs = new URLSearchParams(params).toString();
      url = path + "?" + qs;
    }
    return this._request("GET", url);
  }

  async post(path, body) {
    return this._request("POST", path, body || {});
  }

  async delete(path, body) {
    return this._request("DELETE", path, body || {});
  }

  async put(path, body) {
    return this._request("PUT", path, body || {});
  }
}
